examples/preprocessing/plot_artifact_correction_with_faster.py

```diff
@@ -0,0 +1,119 @@
+"""
+====================================
+EEG artifact correction using FASTER
+====================================
+
+In this example, a variety of metrics are use to detect channels, epochs and
+ICA components that contain artifacts. Rejection and interpolation are used to
+clean the EEG data.
+
+References
+----------
+[1] Nolan H., Whelan R. and Reilly RB. Engemann D. FASTER: fully automated
+    statistical thresholding for EEG artifact rejection. Journal of
+    Neuroscience Methods, vol. 192, issue 1, pp. 152-162, 2010.
+"""
+import mne
+from mne import io
+from mne.preprocessing import (faster_bad_channels, faster_bad_epochs,
+                               faster_bad_channels_in_epochs, ICA,
+                               create_ecg_epochs, create_eog_epochs)
+from mne.datasets import sample
+
+# Load raw data
+data_path = sample.data_path()
+raw_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw.fif'
+event_fname = data_path + '/MEG/sample/sample_audvis_filt-0-40_raw-eve.fif'
+
+raw = io.Raw(raw_fname, preload=True)
+raw.info['bads'] = []  # bads are going to be detected automatically
+events = mne.read_events(event_fname)
+
+# Keep whatever EEG reference the amplifier used for now. After the data is
+# cleaned, we will re-reference to an average reference.
+raw, _ = io.set_eeg_reference(raw, [])
+
+# Highpass filter the EEG, EOG and ECG data to eliminate drifts
+# NOTE: we do not lowpass filter here, because power line noise is useful for
+#       detecting bad channels.
+picks = mne.pick_types(raw.info, meg=False, eeg=True, eog=True, ecg=True)
+raw.filter(0.3, None, method='iir', picks=picks)
+
+# Construct epochs
+event_ids = {'AudL': 1, 'AudR': 2, 'VisL': 3, 'VisR': 4}
+tmin = -0.2
+tmax = 0.5
+picks = mne.pick_types(raw.info, meg=True, eeg=True, eog=True, ecg=True)
+epochs = mne.Epochs(raw, events, event_ids, tmin, tmax, baseline=(None, 0),
+                    preload=True, picks=picks)
+
+# Compute evoked before cleaning, using an average EEG reference
+epochs_before = epochs.copy()
+epochs_before.info['custom_ref_applied'] = False
+epochs_before, _ = io.set_eeg_reference(epochs_before)
```

Comment:
attempted to move it... created circular import... gave up.
Also, average referencing is handled in `io.proj`, so there are some links to other pieces of the `io` module.